src/systems/advanced_lift_drag/AdvancedLiftDrag.cc

```diff
@@ -0,0 +1,825 @@
+  // gzdbg << "alpha: " << this->alpha << "\n";
+  // gzdbg << "beta: " << this->beta << "\n";
+  // gzdbg << "sigma: " << sigma << "\n";
+
+  // Compute control surface effects
+  double CL_ctrl_tot = 0;
+  double CD_ctrl_tot = 0;
+  double CY_ctrl_tot = 0;
+  double Cell_ctrl_tot = 0;
+  double Cem_ctrl_tot = 0;
+  double Cen_ctrl_tot = 0;
+
+
+  for(int i = 0; i < this->num_ctrl_surfaces; i++){
+    double controlAngle = 0.0;
+    if (controlJointPosition_vec[i] && !controlJointPosition_vec[i]->Data().empty())
+    {
+      components::JointPosition *tmp_controlJointPosition = controlJointPosition_vec[i];
+      controlAngle = tmp_controlJointPosition->Data()[0] * 180/M_PI;
+    }
+
+    /* AVL's and Gazebo's direction of "positive" deflection may be different.
+    Future users, keep an eye on this. */
+    CL_ctrl_tot += controlAngle*this->CL_ctrl[i]*this->ctrl_surface_direction[i];
+    CD_ctrl_tot += controlAngle*this->CD_ctrl[i]*this->ctrl_surface_direction[i];
+    CY_ctrl_tot += controlAngle*this->CY_ctrl[i]*this->ctrl_surface_direction[i];
+    Cell_ctrl_tot += controlAngle*this->Cell_ctrl[i]*this->ctrl_surface_direction[i];
+    Cem_ctrl_tot += controlAngle*this->Cem_ctrl[i]*this->ctrl_surface_direction[i];
+    Cen_ctrl_tot += controlAngle*this->Cen_ctrl[i]*this->ctrl_surface_direction[i];
+  }
+
+  // AVL outputs a "CZ_elev", but the Z axis is down. This plugin
+  // uses CL_elev, which is the negative of CZ_elev
+  CL = CL+CL_ctrl_tot;
+  // gzdbg << "Current CL:" << CL << "\n";
+
+  // Compute lift force at cp
+  gz::math::Vector3d lift = (CL * dyn_pres + (this->CLp * (rr*span/2) * half_rho_vel) + (this->CLq * (pr*this->mac/2) * half_rho_vel) + (this->CLr * (yr*span/2) * half_rho_vel)) * (this->area * (-1 * stability_z_axis));
+
+  // Compute CD at cp, check for stall
+  double CD{0.0};
+
+  //Add in quadratic model and a high-angle-of-attack (Newtonian) model
+
+    /* The post stall model used below has two parts. The first part, the
+    (estimated) flat plate drag, comes from the data in Ostowari and Naik,
+    Post-Stall Wind Tunnel Data for NACA 44XX Series Airfoil Sections.
+    https://www.nrel.gov/docs/legosti/old/2559.pdf
+    The second part (0.5-0.5cos(2*alpha)) is fitted using data from Stringer et al,
+    A new 360° airfoil model for predicting airfoil thrust potential in
+    vertical-axis wind turbine designs.
+    https://aip.scitation.org/doi/pdf/10.1063/1.5011207
+    I halved the drag numbers to make sure it would work with my flat plate drag model.
+    */
+
+    /*To estimate the flat plate coefficient of drag, I fit a sigmoid function
+    to the data in Ostowari and Naik. The form I used was:
+    CD_FP = 2/(1+exp(k1+k2*AR)).
+    The coefficients k1 and k2 might need some tuning.
+    I chose a sigmoid because the CD would initially increase quickly with aspect
+    ratio, but that rate of increase would slow down as AR goes to infinity.*/
```

Comment:
Indentation and use `//`
```suggestion
    /* The post stall model used below has two parts. The first part, the
    (estimated) flat plate drag, comes from the data in Ostowari and Naik,
    Post-Stall Wind Tunnel Data for NACA 44XX Series Airfoil Sections.
    https://www.nrel.gov/docs/legosti/old/2559.pdf
    The second part (0.5-0.5cos(2*alpha)) is fitted using data from Stringer et al,
    A new 360° airfoil model for predicting airfoil thrust potential in
    vertical-axis wind turbine designs.
    https://aip.scitation.org/doi/pdf/10.1063/1.5011207
    I halved the drag numbers to make sure it would work with my flat plate drag model.
    */

    /*To estimate the flat plate coefficient of drag, I fit a sigmoid function
    to the data in Ostowari and Naik. The form I used was:
    CD_FP = 2/(1+exp(k1+k2*AR)).
    The coefficients k1 and k2 might need some tuning.
    I chose a sigmoid because the CD would initially increase quickly with aspect
    ratio, but that rate of increase would slow down as AR goes to infinity.*/
```